File: app/src/test/java/com/sakuna63/tumbin/data/util/TestUtils.java
package com.sakuna63.tumbin.data.util;

import android.support.annotation.NonNull;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;

public class TestUtils {

    private static String resolveFilePath(@NonNull String fileName) {
        final String basePath = "./src/test/assets/";
        return basePath + fileName;
    }

    private static InputStream readFile(@NonNull String filePath) {
        try {
            return new FileInputStream(filePath);
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        }
    }

    public static <T> T readJsonAssets(@NonNull ObjectMapper objectMapper,
                                       @NonNull TypeReference<T> type,
                                       @NonNull final String fileName) {
        InputStream in = readFile(resolveFilePath(fileName));
        try {
            return objectMapper.readValue(in, type);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: app/src/test/java/com/sakuna63/tumbin/data/util/TestUtils.kt
package com.sakuna63.tumbin.data.util

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper

import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStream

object TestUtils {

    private fun resolveFilePath(fileName: String): String {
        val basePath = "./src/test/assets/"
        return basePath + fileName
    }

    private fun readFile(filePath: String): InputStream {
        try {
            return FileInputStream(filePath)
        } catch (e: FileNotFoundException) {
            throw RuntimeException(e)
        }

    }

    fun <T> readJsonAssets(objectMapper: ObjectMapper,
                           type: TypeReference<T>,
                           fileName: String): T {
        val `in` = readFile(resolveFilePath(fileName))
        try {
            return objectMapper.readValue<T>(`in`, type)
        } catch (e: IOException) {
            throw RuntimeException(e)
        }

    }
}
